ui/src/components/Views/Configs/MountPointForm.tsx

```diff
@@ -1,137 +1,187 @@
-      ['--rm', '--entrypoint=', '-v', `/${path}:/users`, 'localstack/localstack', 'ls', '/users']);
-
-    if (res.stderr !== '' || res.stdout === '') {
-      ddClient.desktopUI.toast.error(`Error while locating users: ${res.stderr}\n using /tmp as mount point`);
-      setUserState({ loading: false, selectedUser: 'tmp', users: ['tmp'] });
-      setMountPointUser('tmp');
+  const getMountPointPath = (): string => {
+    if(ddClient.host.platform === 'darwin'){
+      return `/Users/${userState.selectedUser || 'loading...'}/Library/Caches/localstack/volume`;
     }
-    const foundUsers = res.stdout.split('\n');
-    foundUsers.pop(); // remove last '' element
-    setUserState({ loading: false, selectedUser: foundUsers[0], users: foundUsers });
+    return `/home/${userState.selectedUser || 'loading...'}/.cache/localstack/volume`;
   };
 
-  useEffect(() => {
-    const execChecks = async () => {
-      if (userState.users.length === 0) {
+  const checkWindowsDistro = async () => {
+    setOsState({ ...osState, loading: true });
 
-        setHasLocalImage({ ...hasLocalImage, checking: true });
+    const res = await ddClient.extension.host?.cli.exec('checkWSLOS.cmd', []);
+    
+    const foundOSs = getOSsFromBinary(res.stdout);
+
+    setOsState({ loading: false, selectedOS: foundOSs[0], OSs: foundOSs });
+    setTriggerUserCheck(!triggerUserCheck);
+  };
+
+  const checkUser = async () => {
+    setUserState({ ...userState, loading: true });
+
+    let res: ExecResult;
+    let foundUsers = [];
+    if (ddClient.host.platform === 'win32') {
+      res = await ddClient.extension.host?.cli.exec('checkUser.cmd', [osState.selectedOS]);
+      foundUsers = getUsersFromBinaryWindows(res.stdout);
+    } else {
+      res = await ddClient.extension.host?.cli.exec('checkUser.sh', []);
+      foundUsers = getUsersFromBinaryUnix(res.stdout);
+    }
+
+    if (res.stderr || !res.stdout) {
+      ddClient.desktopUI.toast.error(`Error while locating users: ${res.stderr} using /tmp as mount point`);
+      setUserState({ loading: false, selectedUser: '../tmp', users: ['tmp'] });
+      setMountPointData('../tmp');
+    }
 
-        const images = await ddClient.docker.listImages() as [DockerImage];
-        const isPresent = images.some(image => image.RepoTags?.at(0) === LATEST_IMAGE);
+    setUserState({ loading: false, selectedUser: foundUsers[0], users: foundUsers });
+  };
 
-        setHasLocalImage({ checking: false, isPresent });
+  const locateMountPoint = async () => {
+    if (ddClient.host.platform === 'win32') {
+      checkWindowsDistro();
+    } else {
+      checkUser();
+    }
+  };
 
-        if (isPresent) {
-          checkHomeDir();
-        } else {
-          setIsPullingImage(true);
-        }
+  useEffect(() => {
+    const execChecks = async () => {
+      if (userState.users.length === 0
+        || (ddClient.host.platform === 'win32' && osState.OSs.length === 0)) {
+        locateMountPoint();
```

Comment:
nit: not sure what the convention is here, but presumably we're missing an indentation here?
```suggestion
          locateMountPoint();
```

Again, should ideally be caught/defined by the linter

Reply:
This is technically not an error since the body is 1 indent more than the if...maybe 
```
if ( x 
|| y ) {
  locateMountPoint();
```
  would be better?